asv_benchmarks/benchmarks/pairwise_distances_reductions.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+from scipy.spatial.distance import cdist
+
+from .common import Benchmark
+
+from sklearn.metrics._pairwise_distances_reduction import (
+    PairwiseDistancesArgKmin,
+    PairwiseDistancesRadiusNeighborhood,
+)
+
+# To run benchmarks defined this file, between for instance your <current_branch>
+# and its fork point with upstream/main, run:
+#
+#             asv continuous -b PairwiseDistancesReductions \
+#             -e `git merge-base --fork-point upstream/main <your_branch>` <your_branch>
+
+
+class PairwiseDistancesReductionsBenchmark(Benchmark):
+
+    param_names = ["n_train", "n_test", "n_features", "metric", "strategy"]
+    params = [
+        [1000, 10_000, int(1e7)],
```

Comment:
As stated in https://github.com/scikit-learn/scikit-learn/pull/24120#pullrequestreview-1066711286 and https://github.com/scikit-learn/scikit-learn/pull/24120#issuecomment-1209549462, I do not think we can include this in asv benchmarks.